src/importV1/index.ts

```diff
@@ -0,0 +1,192 @@
+import * as fs from 'fs';
+import * as stream from 'stream';
+import * as util from 'util';
+
+import { DbBNSName, DbBNSNamespace } from '../datastore/common';
+import { logError, logger } from '../helpers';
+
+import * as split from 'split2';
+import { PgDataStore } from '../datastore/postgres-store';
+import { PoolClient } from 'pg';
+
+const finished = util.promisify(stream.finished);
+const pipeline = util.promisify(stream.pipeline);
+
+class ChainProcessor extends stream.Writable {
+  tag: string = 'chainprocessor';
+  state: string = '';
+  rowCount: number = 0;
+  zhashes: Record<string, string>;
+  db: PgDataStore;
+  client!: PoolClient;
+
+  constructor(db: PgDataStore, zhashes: Record<string, string>) {
+    super();
+    this.zhashes = zhashes;
+    this.db = db;
+    logger.info(`${this.tag}: namespace and name importer starting`);
+  }
+
+  _final() {
+    logger.info(`${this.tag}: namespace and name importer done`);
+  }
+
+  async _write(chunk: any, encoding: string, next: (error?: Error) => void) {
+    const line = (chunk as Buffer).toString();
+
+    /*
+     * Blocks of data inside the chainstate file start with
+     * -----BEGIN SOMETHING-----
+     * and end with
+     * -----END SOMETHING-----
+     *
+     * Extract text between "BEGIN " and the trailing dashes, replace
+     * spaces with underscores, and downcase the strings.
+     */
+
+    if (line.startsWith('-----BEGIN')) {
+      const state = line
+        .slice(line.indexOf(' '))
+        .split('-')[0]
+        .trim()
+        .toLowerCase()
+        .replace(/\s+/, '_');
+
+      // we only care about namespaces/names
+      if (state.startsWith('name')) {
+        this.state = state;
+      } else {
+        this.state = '';
+      }
+    } else if (line.startsWith('-----END')) {
+      if (this.state.startsWith('name') && this.rowCount > 0) {
+        logger.info(`${this.tag}: ${this.state} rows added: ${this.rowCount}`);
+      }
+      this.state = '';
+      this.rowCount = 0;
+    } else if (this.state != '') {
+      this.client = await this.db.pool.connect();
```

Comment:
Connecting and releasing a client each time probably slows this down quite a bit. It should use one client for the full duration of the import.

Reply:
I found it easier to batch up the subdomain objects in a stream reader function, and then perform a bulk insert.  I couldn't figure out how to write something like `batchIterateAsync`, but it works the same way, albeit not as cleanly.  I'll look into it some more.

It does a little over 2000 inserts/second right now.